### ISSUE TITLE
Any hope for a 4MB version? Since write now it's too big. 

### ISSUE THREAD

**Issue by SSadistic** [apollo-ng/luci-theme-darkmatter#6]
When I try to install it, I'm always running out of space.

**Comment by chron0** [comment]
should be not a problem, there are no big assets and it's kept at minimum, let's see if the new approach after #4 is implemented fixes that as well, if not I'll look into it some more in terms of optimizing for size.

**Comment by chron0** [comment]
There is still headroom in the css. Instead of putting the images base64 encoded into the css we can put them down as assets, it will increase loading time but will help with space constraints on 4MB devices. The fonts can also be cut down to carry only what we really need and we can pre gzip jquery.min down to 38kB.

After all, it's just a nasty process, the UI is hard to style and put into something user friendly. There are so many exceptions and hacks to get rid of all empty space, rows, br's and whatnot.

Ideally there should be no LuCI server generated html at all. A small angular2 based web client which communicates via websocket with UCI would be a much more efficient and elegant approach to go into the future.

**Comment by chron0** [comment]
### Source size

Initial release:

    luci-theme-darkmatter $ du -sch . -> 476K

After first optimization:

    luci-theme-darkmatter $ du -sch . -> 188K

### Final Package size

Initial release: 138K
After first optimization: 68K

**Comment by chron0** [comment]
@SSadistic could you please try again and see if you can install it (beta-2) now?

**Comment by weedy** [comment]
https://forum.lede-project.org/t/openwrt-as-default-web-luci-theme/412/59 
I don't know if you've seen all these comments.

**Comment by chron0** [comment]
Yes, unfortunately, there is only talk and no PRs. Anyone can sit down and waste time in forums, bashing other people's attempts to move forward. Only by actually tackling the problems like this jquery dependency and luci under-the-hood workings we can get somewhere. Since it was only just released as a demo to see if people actually would like to have a darker interface for luci (there was none!) - it's only natural that things aren't heavily optimized and refined. The basic goal was to have something like material, that could deploy anywhere where material could run and go on from there.

**Comment by adil1214** [comment]
hi from the future, i just tested it on a 4mb device with only 140kb available after a fresh firmware flash and it took about 84kb of free space.
the only problem i had is when trying to download over https using wget 

> wget https://apollo.open-resource.org/downloads/luci-theme-darkmatter_0.2-beta-2_all.ipk
wget: SSL support not available, please install one of the libustream-ssl-* libraries as well as the ca-bundle and ca-certificates packages.

as a workaround to this, i downloaded the package to my machine first then served it over an http server.
